climada_petals/hazard/test/test_wildfire.py

```diff
@@ -274,20 +274,23 @@ def test_centroids_pass(self):
         wf = WildFire()
         firms = wf._clean_firms_df(TEST_FIRMS)
         centroids = wf._firms_centroids_creation(firms, 0.375/ONE_LAT_KM, 1/2)
-        self.assertEqual(centroids.meta['width'], 144)
-        self.assertEqual(centroids.meta['height'], 138)
-        self.assertAlmostEqual(centroids.meta['transform'][0], 0.006749460043196544)
-        self.assertAlmostEqual(centroids.meta['transform'][1], 0.0)
-        self.assertTrue(centroids.meta['transform'][2]<= firms.longitude.min())
-        self.assertAlmostEqual(centroids.meta['transform'][3], 0.0)
-        self.assertAlmostEqual(centroids.meta['transform'][4], -centroids.meta['transform'][0])
-        self.assertTrue(centroids.meta['transform'][5] >= firms.latitude.max())
-        self.assertTrue(firms.latitude.max() <= centroids.total_bounds[3])
-        self.assertTrue(firms.latitude.min() >= centroids.total_bounds[1])
-        self.assertTrue(firms.longitude.max() <= centroids.total_bounds[2])
-        self.assertTrue(firms.longitude.min() >= centroids.total_bounds[0])
+        centroids_meta = centroids.get_meta()
+        self.assertEqual(centroids_meta['width'], 144)
+        self.assertEqual(centroids_meta['height'], 138)
+        self.assertAlmostEqual(centroids_meta['transform'][0], 0.006749460043196544)
+        self.assertAlmostEqual(centroids_meta['transform'][1], 0.0)
+        self.assertLessEqual(centroids_meta['transform'][2], firms.longitude.min())
+        self.assertAlmostEqual(centroids_meta['transform'][3], 0.0)
+        self.assertAlmostEqual(centroids_meta['transform'][4], -centroids_meta['transform'][0])
+        self.assertGreaterEqual(centroids_meta['transform'][5], firms.latitude.max())
+        self.assertLessEqual(firms.latitude.max(), centroids.total_bounds[3])
+        # TODO: check whether allowing for a tolerance is valid here
+        TOLERANCE = 0.005
```

Comment:
How to check that? 🤔

Reply:
Actually, the method [Centroids.from_pnt_bounds](https://github.com/CLIMADA-project/climada_petals/blob/9f48d3fdae8d11d4a4c61212ff1a469d9cf1ef69/climada_petals/hazard/wildfire.py#L658) should produce exactly the same latitude and longitude values in both versions of the code. 

I think the point here is that we are now only using the centre of the raster pixels instead of the raster itself, there might be a half-pixel difference in the bounds. I am not sure to understand what this test is supposed to check... Why is the tolerance set to 0.005? Why should there be a tolerance? O really have to dig into this code to understand this.

Reply:
the tolerance is an artifact that I introduced simply to make the test pass, suspecting that the reason for failure may be something like what you mention - and that it does not matter.

Reply:
aaaa got it! I will try to debug and find the reason.

Reply:
So, the change has to do with the pixel definition. We have that 

`new_centroids.total_bounds[1] - 0.375/ONE_LAT_KM= old_centroids.total_bounds[1]`

This almost makes sense, as `0.375/ONE_LAT_KM` is the data raster resolution. But I would have expected a factor of one-half... However, the method `wf._firms_centroids_creation(firms, 0.375/ONE_LAT_KM, 1/2)` applies another factor to obtain the final raster resolution, here equal exactly to 1/2, which thus cancels the other one out.

Reply:
Solution suggested in [116e2e8](https://github.com/CLIMADA-project/climada_petals/pull/122/commits/116e2e83d79c292735f65a74a7bc65c518c842aa)

Reply:
🎉 excellent analysis! I'm still wondering though - shall we keep the test ast it is, do we change it or should we even got back and change the implementation?

Reply:
ok. 116e2e8 answers the question 😄